src/test/java/org/logstash/beats/ServerTest.java

```diff
@@ -34,9 +42,43 @@ public void setUp() {
     @Test
     public void testServerShouldTerminateConnectionIdleForTooLong() throws InterruptedException {
         int inactivityTime = 3; // in seconds
+        int concurrentConnections = 10;
+
+        final Random random = new Random();
+
+        final CountDownLatch latch = new CountDownLatch(concurrentConnections);
 
         final Server server = new Server(host, randomPort, inactivityTime);
+        final AtomicBoolean otherCause = new AtomicBoolean(false);
+        server.setMessageListener(new MessageListener() {
+            public void onNewConnection(ChannelHandlerContext ctx) {
+                // Make sure connection is closed on exception too.
+                if (random.nextInt(10) < 1) {
+                    throw new RuntimeException("Dummy");
+                }
+            }
+
+            @Override
+            public void onConnectionClose(ChannelHandlerContext ctx) {
+                latch.countDown();
+            }
+
+            @Override
+            public void onNewMessage(ChannelHandlerContext ctx, Message message) {
+                // Make sure connection is closed on exception too.
```

Comment:
what side effect exists if a connection is not closed? I don't see anything explicit in the assertions in this test?